tools/xnvme.c

```diff
@@ -623,6 +623,34 @@ sub_library_info(struct xnvmec *XNVME_UNUSED(cli))
 	return err;
 }
 
+static int
+sub_dsm(struct xnvmec *cli)
+{
+	struct xnvme_dev *dev = cli->args.dev;
+	struct xnvme_cmd_ctx ctx = xnvme_cmd_ctx_from_dev(dev);
+	uint32_t nsid = cli->args.nsid;
+	uint32_t nr = cli->args.nr;
+	uint32_t ad = cli->args.ad;
```

Comment:
The ``ad``, ``idw``, and ``idr`` should probably just be bools.

Reply:
Done